src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1326,10 +1326,11 @@ public override bool IsTupleCompatible(out int tupleCardinality)
 
                     do
                     {
+                        Debug.Assert(!typeToCheck.IsDefinition);
```

Comment:
We're checking `this.IsDefinition` at line 1321. Is it possible to use a generic definition as a type argument?